### PR TITLE
Suppress a bunch of unnecessary test output

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

## Summary

This test floods the logs without providing any meaningful value or insights.  It makes looking at jenkins or actions results more troublesome than necessary.  This redirects the output to another buffer.

It doesn't seem to have any impact on the timing results, either:

Before:
```
Using 32 to log in total 10000 log entries to 32.result.csv
0 the worst thread latency was:4 ms  (4659 us)] 
1 the worst thread latency was:5 ms  (5373 us)] 
2 the worst thread latency was:4 ms  (4225 us)] 
3 the worst thread latency was:4 ms  (4823 us)] 
4 the worst thread latency was:6 ms  (6415 us)] 
5 the worst thread latency was:6 ms  (6038 us)] 
6 the worst thread latency was:6 ms  (6863 us)] 
7 the worst thread latency was:5 ms  (5238 us)] 
8 the worst thread latency was:5 ms  (5238 us)] 
9 the worst thread latency was:5 ms  (5478 us)] 
10 the worst thread latency was:7 ms  (7307 us)] 
11 the worst thread latency was:5 ms  (5909 us)] 
12 the worst thread latency was:4 ms  (4853 us)] 
13 the worst thread latency was:4 ms  (4619 us)] 
14 the worst thread latency was:5 ms  (5235 us)] 
15 the worst thread latency was:5 ms  (5163 us)] 
16 the worst thread latency was:4 ms  (4508 us)] 
17 the worst thread latency was:6 ms  (6340 us)] 
18 the worst thread latency was:6 ms  (6755 us)] 
19 the worst thread latency was:6 ms  (6466 us)] 
20 the worst thread latency was:8 ms  (8261 us)] 
21 the worst thread latency was:5 ms  (5793 us)] 
22 the worst thread latency was:5 ms  (5150 us)] 
23 the worst thread latency was:4 ms  (4631 us)] 
24 the worst thread latency was:5 ms  (5080 us)] 
25 the worst thread latency was:6 ms  (6668 us)] 
26 the worst thread latency was:4 ms  (4814 us)] 
27 the worst thread latency was:4 ms  (4888 us)] 
28 the worst thread latency was:6 ms  (6234 us)] 
29 the worst thread latency was:4 ms  (4519 us)] 
30 the worst thread latency was:4 ms  (4603 us)] 
31 the worst thread latency was:4 ms  (4643 us)] 
Total time :459 ms (459614 us)
Average time: 45.9614 us
```

After:
```
Using 32 to log in total 10000 log entries to 32.result.csv
0 the worst thread latency was:4 ms  (4090 us)] 
1 the worst thread latency was:4 ms  (4886 us)] 
2 the worst thread latency was:4 ms  (4422 us)] 
3 the worst thread latency was:4 ms  (4958 us)] 
4 the worst thread latency was:6 ms  (6431 us)] 
5 the worst thread latency was:4 ms  (4714 us)] 
6 the worst thread latency was:5 ms  (5029 us)] 
7 the worst thread latency was:4 ms  (4306 us)] 
8 the worst thread latency was:5 ms  (5436 us)] 
9 the worst thread latency was:4 ms  (4923 us)] 
10 the worst thread latency was:4 ms  (4976 us)] 
11 the worst thread latency was:4 ms  (4794 us)] 
12 the worst thread latency was:5 ms  (5419 us)] 
13 the worst thread latency was:5 ms  (5248 us)] 
14 the worst thread latency was:5 ms  (5467 us)] 
15 the worst thread latency was:5 ms  (5828 us)] 
16 the worst thread latency was:5 ms  (5626 us)] 
17 the worst thread latency was:5 ms  (5267 us)] 
18 the worst thread latency was:5 ms  (5593 us)] 
19 the worst thread latency was:4 ms  (4397 us)] 
20 the worst thread latency was:4 ms  (4380 us)] 
21 the worst thread latency was:4 ms  (4759 us)] 
22 the worst thread latency was:5 ms  (5477 us)] 
23 the worst thread latency was:5 ms  (5262 us)] 
24 the worst thread latency was:4 ms  (4901 us)] 
25 the worst thread latency was:5 ms  (5201 us)] 
26 the worst thread latency was:5 ms  (5202 us)] 
27 the worst thread latency was:4 ms  (4620 us)] 
28 the worst thread latency was:4 ms  (4581 us)] 
29 the worst thread latency was:6 ms  (6674 us)] 
30 the worst thread latency was:5 ms  (5780 us)] 
31 the worst thread latency was:6 ms  (6581 us)] 
Total time :458 ms (458197 us)
Average time: 45.8197 us
```

Signed-off-by: Michael Carroll <michael@openrobotics.org>

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

**Comment by mjcarroll** [comment]
Looks like you aren't supposed to use the `internal` methods from `gtest` (which makes sense).  I think I will write a small utility class in the `testing` module for this

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

File: testing/include/gz/common/testing/RedirectConsoleStream.hh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/// \brief Redirect standard error to a test-specific temporary path
+inline RedirectConsoleStream RedirectStderr() 
+{
+  auto path = gz::common::testing::TempPath("stderr.out");
+  return RedirectConsoleStream(StreamSource::STDERR, path);
+}
+
+
+}  // namespace gz::common::testing
```
Comment (ahcorde) [inline testing/include/gz/common/testing/RedirectConsoleStream.hh:77]:
```suggestion
}
}  // namespace gz::common::testing
```

File: testing/src/RedirectConsoleStream.cc [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+
+#include <cstring>
```
Comment (ahcorde) [inline testing/src/RedirectConsoleStream.cc:18]: alpahbetize

File: testing/src/RedirectConsoleStream.cc [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+#include <cstdio>
+#include <cstdlib>
+
+#include "gz/common/testing/RedirectConsoleStream.hh"
+#include "gz/common/Console.hh"
+#include <gz/common/testing/TestPaths.hh>
+
+#include <fstream>
```
Comment (ahcorde) [inline testing/src/RedirectConsoleStream.cc:26]: with the other standard libraries

File: testing/src/RedirectConsoleStream.cc [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    GZ_UTILS_WARN_IGNORE__DEPRECATED_DECLARATION;
+    /// Restore the orignal source file descriptor
+    if (this->originalFd != -1)
+    {
+      fflush(nullptr);
+      dup2(this->originalFd, this->fd);
+      close(this->originalFd);
+      this->originalFd = -1;
+    }
+    GZ_UTILS_WARN_RESUME__DEPRECATED_DECLARATION;
```
Comment (ahcorde) [inline testing/src/RedirectConsoleStream.cc:78]: indentation

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by mjcarroll** [comment]
Windows failures are the known GDAL problems.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**